src/segmented_buffer.rs

```diff
@@ -348,13 +471,7 @@ impl<F> std::io::Write for SegmentedPoolBuf<F, BytesMut, AllocBytesMutFn> {
     }
 }
 
-impl AsyncWrite
-    for SegmentedPoolBuf<
-        Pin<Box<dyn Future<Output = Option<Reusable<BytesMut>>> + std::marker::Send>>,
-        BytesMut,
-        AllocBytesMutFn,
-    >
-{
+impl AsyncWrite for SegmentedPoolBuf<BufFut, Buffer, AllocBufferFn> {
```

Comment:
Can the call for `this.pool.expand()` in an async env allocate without bounds?

Reply:
It only allocates when it is polling the buffer future, which only created if the pool is below max size